### ISSUE TITLE
Add support for writing configuration files

### ISSUE THREAD

**Issue by filips123** [hassankhan/config#107]
Add support for writing configuration files. This would be useful so developers could make user-friendly installers for configuration files.

For INI, XML, JSON, YAML and [other formats](https://github.com/hassankhan/config/issues/106), this should not be hard, because they already support writing. For PHP files, this could be harder.

**Example of writing config from start:**
```php
// Instantiation
$conf = new Config();

// Setting value using key
$conf->set('debug', true);
$conf['debug'] = true;

// Setting value using nested key
$conf->set('security.secret', 'eXaMpLe');
$conf['security.secret'] = 'eXaMpLe';
$conf['security']['secret'] = 'eXaMpLe';

// Setting all values from array
$settings = array(
    'debug' => true,
    'security' => array(
        'secret' => 'eXaMpLe',
        'password' => '1234'
    ),
    'app.timeout' => 3000,
    'app.name' => 'Website'
);
$conf->set('', $settings);
$conf->all() = $settings;
$conf = $settings;

// Writing to file
$conf->write('config.json');
```

**Example of writing using existing config:**
```php
// Instantiation
$conf = new Config('config.json');

// Setting value using key
$conf['debug'] = true;

// Writing to existing file (file should now have all previous settings and new ones)
$conf->write();

// Writing to new file (file should now have all previous settings from other file and new ones)
$conf->write('config.yml');
```

**Comment by filips123** [comment]
@DavidePastore @hassankhan I'm working on writing support in [my repository](https://github.com/filips123/config/tree/writing). Currently, the result of writing would be minified (for example one file for JSON). Is this OK or should the result be not minified?

**Comment by DavidePastore** [comment]
@filips123 I would prefer to have the chance to choose (using configuration) if I want minified output or not.

**Comment by filips123** [comment]
@DavidePastore I currently don't have much time, but I will continue implementing writing support next month. Problem is that I would have to change, add and rename methods in parsers and other classes so this would be breaking change and new major version (3.0.0). Is that OK? Would it be possible to do this without breaking changes?

**Comment by DavidePastore** [comment]
@filips123 I think you can do it without breaking changes but by only adding new methods for the purpose. If you already have something in mind you can share it with us (cc @hassankhan ) so we could discuss about it. Anyway it could make sense to create a new major version for this kind of improvement.

**Comment by filips123** [comment]
@DavidePastore I created something in [my repository](https://github.com/filips123/config/tree/writing). However, those changes are old and I will need to rewrite them because of git conflicts with new updates.

**Comment by DavidePastore** [comment]
@filips123 I like the idea behind it and the way you did it. The breaking change you were talking about before was the `parse`-> `decode` refactoring, right?

**Comment by filips123** [comment]
> I like the idea behind it and the way you did it. 

Thank you. I need to fix all git conflicts. I will then also provide support for writer configuration (https://github.com/hassankhan/config/issues/107#issuecomment-460059233).

> The breaking change you were talking about before was the parse-> decode refactoring, right?

Yes.

**Comment by filips123** [comment]
Fixed with #122 in [2.1.0](https://github.com/hassankhan/config/tree/2.1.0).
